### PR TITLE
Show ressources

### PR DESCRIPTION
**Pull request by Flashacker-Neo**
Add mod for display all resources on map with a shortcut.

### CODE REVIEW

File: locale/en.json [hunk 1]
```diff
@@ -67,7 +67,9 @@
             "estimator": "Estimating spell damage in battle",
             "hide-mount": "Hide mounts during fights",
             "fightchronometer" : "Show combats chronometer",
-            "zaapsearchfilter" : "Show zaap search filter"
+            "zaapsearchfilter" : "Show zaap search filter",
+            "show-resources" : "Show map resources",
+            "show-resources-shortcut" : "Shortcut for show/hide map resources"
```
Comment (Maryannah) [inline locale/en.json:72]: `to show/hide`

File: src/app/core/mods/showResources/resources.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    private resources: {elemId: number, state: State}[] = [];
+
+    constructor(resource: Resource) {
+        this.name = resource.name;
+        this.typeId = resource.elementTypeId;
+        this.addOrUpdateResource(resource);
+    }
+
+    public addOrUpdateResource(resource: Resource) {
+        if (this.resources.length > 0 && this.resources.find(r => r.elemId == resource.elementId) != undefined) {
```
Comment (Maryannah) [inline src/app/core/mods/showResources/resources.ts:13]: `if (this.resources.length && this.resources.find(r => r.elemId == resource.elementId)) {`

Comment (Maryannah) [reply src/app/core/mods/showResources/resources.ts:13]: Event better 

```
const ref = this.resources.find(r => r.elemId == resource.elementId);
if (ref) ref.state = resource.elementState;
else this.resources.push({elemId: resource.elementId, state: resource.elementState})
```

File: src/app/core/mods/showResources/resources.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            this.resources.find(r => r.elemId == resource.elementId).state = resource.elementState;
+        } else {
+            this.resources.push({elemId: resource.elementId, state: resource.elementState});
+        }
+    }
+
+    public getName(): string {
```
Comment (Maryannah) [inline src/app/core/mods/showResources/resources.ts:20]: name seems to be a constant, so need to use a getter : instead, make the property public and readonly

File: src/app/core/mods/showResources/showResources.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                .resource-item p {
+                    margin: 0;
+                }
+
+                .resource-item img {
+                    width: 35px;
+                }
+            `;
+
+            this.wGame.document.getElementsByTagName('head')[0].appendChild(resourcesBoxCss);
```
Comment (Maryannah) [inline src/app/core/mods/showResources/showResources.ts:55]: `this.wGame.document.querySelector('head').appendChild(resourcesBoxCss);`

**Review by Maryannah — CHANGES_REQUESTED** [review]

**Review by JulienCoutault — APPROVED** [review]

**Pull request merged**